io_scene_nif/modules/nif_export/property/texture/__init__.py

```diff
@@ -53,46 +53,45 @@ def __init__(self):
         self.texture_anim = TextureAnimation()
         self.b_mat = None
 
-        # todo [texture] refactor texture grabbing to use the dict below
         self.slots = {}
         self._reset_fields()
-        self.b_diffuse_slot = None
-        self.b_glow_slot = None
-        self.b_bump_slot = None
-        self.b_normal_slot = None
-        self.b_gloss_slot = None
-        self.b_dark_slot = None
-        self.b_detail_slot = None
-        self.b_decal_0_slot = None
-        self.b_ref_slot = None
-        self.has_alpha_texture = False
 
     def _reset_fields(self):
-        self.b_diffuse_slot = None
-        self.b_glow_slot = None
-        self.b_bump_slot = None
-        self.b_normal_slot = None
-        self.b_gloss_slot = None
-        self.b_dark_slot = None
-        self.b_detail_slot = None
-        self.b_decal_0_slot = None
-        self.b_ref_slot = None
-        self.has_alpha_texture = False
         self.slots = {
-            "Diffuse": None,
+            "Base": None,
             "Dark": None,
             "Emit": None,
             "Glow": None,
             "Gloss": None,
             "Specular": None,
             "Normal": None,
-            "Bump": None,
+            "Bump Map": None,
             "Detail": None,
-            "Decal0": None,
-            "Decal1": None,
+            "Decal 0": None,
+            "Decal 1": None,
             "Decal2": None,
         }
 
+    def get_uv_node(self, b_texture_node):
+        # check if a node is plugged into the b_texture_node's vector input
+        links = b_texture_node.inputs[0].links
+        if not links:
+            return 0
+        uv_node = links[0].from_node
+        if isinstance(uv_node, bpy.types.ShaderNodeUVMap):
+            uv_name = uv_node.uv_map
+            try:
+                # ignore the "UV" prefix
+                return int(uv_name[2:])
```

Comment:
Will this always be UV prefixed.
Is this consistent for both when a uv map is created through import of existing mesh and a user created one.

Reply:
We should just enforce this or any naming convention, and make warn the users if their uv names don't stick to it. We could of course index into the list of mesh uv layer names, but that seems to be more complicated than needed. A lot of the mesh - material entanglement resulted from devs being too cautious.

Reply:
Yeah I can understand that. 
There is also the balance between expecting it to exist and not, if it is a user issue or optional.